pkg/apiserver/rest/spy_client.go

```diff
@@ -146,3 +183,31 @@ func (c *spyLegacyStorageClient) Delete(ctx context.Context, name string, delete
 	klog.Info("method: LegacyStorage.Delete")
 	return nil, false, nil
 }
+
+type dummyList struct {
```

Comment:
I'm using the `Pod` resource from `"k8s.io/apiserver/pkg/apis/example"` for dummy structs. I think we could do the same in here so we don't have to create a dummy resource by ourselves?

Reply:
I would like to keep the dummy resources for now. As discussed on Slack, we're planning to refactor the testing anyways so we could reevaluate then. What I like about the dummy resources is that they are more generic than `Pod` and if needed we can extend them.

Reply:
make sense, we can leave this to the test iteration next 😃